text/0111-element-modifier.md

```diff
@@ -0,0 +1,226 @@
+- Start Date: 2016-01-23
+- RFC PR: https://github.com/emberjs/rfcs/pull/111
+- Ember Issue:
+
+# Summary
+
+As an outcome of [RFC 100](https://github.com/emberjs/rfcs/pull/100#issuecomment-172427565)
+it was decided that element-space helpers should be added to Ember as a public
+API.
+
+In this example, `add-event-listener` fits the syntax of an element modifier:
+
+```hbs
+<span {{add-event-listener 'click' (action 'save')}}>Save</span>
+```
+
+The implementation of `add-event-listener` would have several hooks called during
+rendering, similar to the rendering hooks fired on a component. Unlike a
+component, there is no template/layout for an element modifier. Unlike a
+helper, an element modifier does not return a value.
+
+# Motivation
+
+Element modifiers allow for user-space implementation of event listeners,
+style, and animation tooling that does not exist in Ember today. It brings
+back some functionality removed from the framework in Ember 2.0 (when
+various intimate helper APIs were removed).
+
+The introduction of this API will likely result in the proliferation of
+one or several popular addons for managing element event listeners, style
+and animation.
+
+# Detailed design
+
+### Invocation
+
+An element modifier is invoked in "element space". This is the space between
+`<` and `>` opening an HTML tag. For example:
+
+```hbs
+<button {{flummux}}></button>
+<span {{whipperwill 'carrot'}}><i>Some DOM</i></span>
+<b {{crum bing='whoop'}} zip="bango">Hm...</b>
+```
+
+Element modifiers may be invoked with params or hash arguments.
+
+### Definition and lookup
+
+A basic element modifier is defined with the type of `element-modifier`. For
+example these paths would be global element modifiers in an application:
+
+```
+Classic paths:
+
+  app/element-modifiers/flummux.js
+  app/element-modifiers/whipperwill.js
+
+Pods paths:
+
+  app/flummux/element-modifier.js
+  app/whipperwill/element-modifier.js
+```
+
+Element modifiers, like component and helpers, are eligible for local lookup.
+For example:
+
+```
+Pods paths:
+
+  app/posts/index/element-modifiers/flummux.js
+```
+
+The element modifier class is a default export from these files. For example:
+
+```js
+import Ember from 'ember';
+
+export default Ember.ElementModifier.extend({});
+```
+
+### Hooks
+
+During rendering and teardown of a target element, any attached element
+modifiers will execute a series of hooks. These hooks are:
+
+* `willInsertElement` (only upon initial render)
```

Comment:
what is the use case for this with an element modifier?  I was thinking these would be scope to didInsertElement and willDestroyElement